Loop/Managers/RemoteDataServicesManager.swift

```diff
@@ -6,165 +6,362 @@
 //  Copyright © 2016 Nathan Racklyeft. All rights reserved.
 //
 
+import os.log
 import Foundation
 import LoopKit
 
-final class RemoteDataServicesManager: CarbStoreSyncDelegate {
+protocol RemoteDataServicesManagerDelegate: AnyObject {
 
-    private unowned let deviceDataManager: DeviceDataManager
+    var carbStore: CarbStore? { get }
 
-    private var remoteDataServices = [RemoteDataService]()
+    var doseStore: DoseStore? { get }
 
-    private var lastSettingsUpdate: Date = .distantPast
+    var glucoseStore: GlucoseStore? { get }
 
-    private let log = DiagnosticLog(category: "RemoteDataServicesManager")
+    var settingsStore: SettingsStore? { get }
 
-    init(deviceDataManager: DeviceDataManager) {
-        self.deviceDataManager = deviceDataManager
+    var statusStore: StatusStore? { get }
 
-        NotificationCenter.default.addObserver(self, selector: #selector(loopCompleted(_:)), name: .LoopCompleted, object: deviceDataManager.loopManager)
-        NotificationCenter.default.addObserver(self, selector: #selector(loopDataUpdated(_:)), name: .LoopDataUpdated, object: deviceDataManager.loopManager)
-    }
+}
+
+final class RemoteDataServicesManager {
+
+    public typealias RawState = [String: Any]
+
+    public weak var delegate: RemoteDataServicesManagerDelegate?
```

Comment:
This, and the RemoteDataServicesManagerDelegate protocol do not seem to be being used.

Reply:
These will be used for the last feature I mentioned in my latest PR comment. Specifically, for fully synchronizing a remote data service when it is first added to Tidepool Loop. There may not be any data updates for the various stores for some time so the initial sync has to walk through each of the stores and synchronize using a nil anchor.

Reply:
Done.

Reply:
I don't think accessing the stores should be done via delegate.  Would need to see the actual interaction to make a suggestion for a better alternative.  Can they just be removed for this PR, and added when needed?

Reply:
It is already integrated. If you take a look here, when a remote data service (e.g. Nightscout) is initially added:

https://github.com/tidepool-org/Loop/pull/32/files#diff-08dca496735f00d680f0f9aa315696d1R47

the `uploadExistingData(to:)` function is invoked:

https://github.com/tidepool-org/Loop/pull/32/files#diff-08dca496735f00d680f0f9aa315696d1R63

and that iterates through the stores and uploads all of the data to the newly added remote data service.

If we don't do it this way, then the previous data only gets uploaded when each data type (i.e. carb, dose, glucose, pumpEvent, settings, status) has new data. This will yield some data types (e.g. glucose) being available online well before other data types (e.g. settings).

So, I see there are 3 possible solutions:
1) Upload all previous data for all data types right after the remote data service is added.
2) All previous data is uploaded for each data type when new data is available for that data type. 
3) Avoid this situation altogether. When a remote data service is first added, grab and store the current query anchors for each data type so that NO previous data is uploaded at all.

However, solutions #2 and #3 have the problem of not getting at least one settings or status data to the remote data service right after it is added.

If you are okay with #1, but not okay with delivering the stores via delegate, the problem is one of initialization order. I can look at that again, but IIRC it was an issue of the stores requiring the RemoteDataServicesManager as a delegate (to notify when new data is available) and the remote data services (via RemoteDataServiceManager) not having the stores fully constructed to pass during initialization.

Thoughts?

Reply:
Yeah, it's just the odd way of providing access to the stores. This goes back to my comments about DeviceDataManager, and who owns the stores, and how they're provided to the other managers.  A top level manager should own them, and be responsible for creating them, and should pass them during initialization to the other managers.